src/joining.h

```diff
@@ -22,101 +22,40 @@
 #include "hash-join/join_compute_api.h"
 #include "sort-join.cuh"
 
-// transpose
-template<typename size_type, typename joined_type>
-void pairs_to_decoupled(mgpu::mem_t<size_type> &output, const size_type output_npairs, joined_type *joined, mgpu::context_t &context, bool flip_indices)
```

Comment:
Shouldn't the prototype still live in the header, even if the implementation is moved?

Reply:
The only place the function is used is in the API. If we leave it in the header file it could cause some weird C include header guards to pop as the header file is including join_compute_api.h